src/app-logic/url-handling.js

```diff
@@ -552,3 +552,64 @@ if (Object.keys(_upgraders).length - 1 !== CURRENT_URL_VERSION) {
     new upgrader, make sure and bump the CURRENT_URL_VERSION variable.
   `);
 }
+
+function getStackIndexFromOldJSCallNodePath(
+  thread: Thread,
+  oldCallNodePath: CallNodePath
+): IndexIntoStackTable | null {
+  const { stackTable, funcTable, frameTable } = thread;
+  const stackMatchesCallNodePath = new Set();
+  const stackIndexDepth: Map<IndexIntoStackTable | null, number> = new Map();
+  stackIndexDepth.set(null, -1);
+
+  for (let stackIndex = 0; stackIndex < stackTable.length; stackIndex++) {
+    const prefix = stackTable.prefix[stackIndex];
+    const frameIndex = stackTable.frame[stackIndex];
+    const funcIndex = frameTable.func[frameIndex];
+    const isJS = funcTable.isJS[funcIndex];
+    const doesPrefixMatchCallNodePath =
+      prefix === null || stackMatchesCallNodePath.has(prefix);
+
+    if (!doesPrefixMatchCallNodePath) {
+      continue;
+    }
+    const prefixStackDepth = ensureExists(
+      stackIndexDepth.get(prefix),
+      'Unable to find the stack depth for a prefix'
+    );
+    const depth = prefixStackDepth + 1;
+
+    if (isJS && oldCallNodePath[depth] === funcIndex) {
+      // This is a JS frame, and it matches the CallNodePath.
+      if (depth === oldCallNodePath.length - 1) {
+        // This is the stack index that we are looking for.
+        return stackIndex;
+      }
+      stackMatchesCallNodePath.add(stackIndex);
+      stackIndexDepth.set(stackIndex, depth);
+    } else {
+      // Any non-JS stack potentially matches, add it here.
```

Comment:
I'd expand a bit more on this comment.
Maybe something like:
```js
// Any non-JS stack potentially matches, because they're skipped in the JS call node path.
// Let's add it here using the previous depth.
```